components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -387,6 +391,42 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
                 }
             }
 
+            // Break the flow if the authenticated subject identifier or user id is null.
+            if (StringUtils.isBlank(sequenceConfig.getAuthenticatedUser().getAuthenticatedSubjectIdentifier())) {
+                if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                    LoggerUtils.triggerDiagnosticLogEvent(
+                            FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, null, LogConstants.FAILED,
+                            ERROR_WHILE_CONCLUDING_AUTHENTICATION_SUBJECT_ID_NULL.getMessage(),
+                            FrameworkConstants.LogConstants.ActionIDs.HANDLE_AUTH_REQUEST, null);
+                }
+                throw new PostAuthenticationFailedException(
+                        ERROR_WHILE_CONCLUDING_AUTHENTICATION_SUBJECT_ID_NULL.getCode(),
+                        ERROR_WHILE_CONCLUDING_AUTHENTICATION_SUBJECT_ID_NULL.getMessage());
+            }
+            try {
+                if (StringUtils.isBlank(sequenceConfig.getAuthenticatedUser().getUserId())) {
+                    if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                        LoggerUtils.triggerDiagnosticLogEvent(
+                                FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, null, LogConstants.FAILED,
+                                ERROR_WHILE_CONCLUDING_AUTHENTICATION_USER_ID_NULL.getMessage(),
+                                FrameworkConstants.LogConstants.ActionIDs.HANDLE_AUTH_REQUEST, null);
+                    }
+                    throw new PostAuthenticationFailedException(
+                            ERROR_WHILE_CONCLUDING_AUTHENTICATION_USER_ID_NULL.getCode(),
+                            ERROR_WHILE_CONCLUDING_AUTHENTICATION_USER_ID_NULL.getMessage());
+                }
+            } catch (UserIdNotFoundException e) {
+                if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                    LoggerUtils.triggerDiagnosticLogEvent(
```

Comment:
Don't we need to log "e.getmessage" to understand the real reason for getting this exception? Otherwise, there is no difference in the diagnostic logs when the user id is blank or with the UserIdNotFoundException

Reply:
It will be the same error. Had to handle it differently as the `AuthenticatedUser.getUserId()` method is throwing it as an exception